presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorPageSourceProvider.java

```diff
@@ -42,4 +44,14 @@ default ConnectorPageSource createPageSource(
     {
         throw new UnsupportedOperationException("createPageSource() must be implemented");
     }
+
+    default ConnectorTableHandle applyDynamicFilter(
```

Comment:
I wonder if it just wouldn't be simpler to pass dynamic filter directly to `createPageSource`. `ConnectorTableHandle` feels more like a planning object, but in `LocalExecutionPlanner` we are already in execution part.

Reply:
Sounds good, will add a new `createPageSource` overload (accepting the dynamic filter as the last argument), IIUC.

Reply:
https://github.com/prestosql/presto/commit/4561c62d7d5b26a2fa12c623d4f279daac7360ea#diff-4dc65ea280f82471ef8da2070c3b9768